src/components/tx-flow/flows/NestedTxSuccessScreen/index.tsx

```diff
@@ -0,0 +1,107 @@
+
+import css from './styles.module.css'
+import Link from 'next/link'
+import { AppRoutes } from '@/config/routes'
+import { useAppSelector } from '@/store'
+import ExternalLink from '@/components/common/ExternalLink'
+
+type Props = {
+  txId: string
+}
+const NestedTxSuccessScreen = ({ txId }: Props) => {
+  const pendingTx = useAppSelector((state) => (txId ? selectPendingTxById(state, txId) : undefined))
+  const addressBook = useAddressBook()
+
+  if (pendingTx?.status !== PendingStatus.NESTED_SIGNING) {
+    return <ErrorMessage>No transaction data found</ErrorMessage>
+  }
+
+  return (
+    <Container
+      component={Paper}
+      disableGutters
+      sx={{
+        textAlign: 'center',
+        maxWidth: `${900 - 75}px`, // md={11}
+      }}
+      maxWidth={false}
+    >
+      <Box padding={3} mt={3} display="flex" flexDirection="column" alignItems="center" gap={2}>
+        <Box className={css.icon}>
+          <SvgIcon component={NestedSafeIcon} inheritViewBox fontSize="large" alt="Nested Safe" />
+        </Box>
+        <Typography data-testid="transaction-status" variant="h6" marginTop={2} fontWeight={700}>
+          A nested transaction was created
+        </Typography>
+        <Typography variant="body2" mb={3}>
+          Once confirmed and executed this signer transaction will confirm the child Safe&apos;s transaction.
+        </Typography>
+        <Stack spacing={2} width="70%">
+          <Box display="flex" flexDirection="column" alignItems="start" gap={1}>
+            <Typography variant="body2" color="text.secondary">
+              Parent Safe
+            </Typography>
+            <EthHashInfo
+              address={pendingTx.signerAddress}
+              name={addressBook[pendingTx.signerAddress]}
+              isAddressBookName={Boolean(addressBook[pendingTx.signerAddress])}
+              shortAddress={false}
+            />
+          </Box>
+          <Stack direction="row" spacing={2} alignItems="center" pl={1}>
+            <SvgIcon component={ArrowDownIcon} fontSize="medium" color="border" inheritViewBox />
+            <Typography
+              component="code"
+              variant="body2"
+              color="primary.light"
+              sx={{
+                backgroundColor: 'background.main',
+                px: 1,
+                py: 0.5,
+                borderRadius: 0.5,
+                fontFamily: 'monospace',
+                whiteSpace: 'nowrap',
+              }}
+            >
+              approveHash
+            </Typography>
+          </Stack>
+          <Box display="flex" flexDirection="column" alignItems="start" gap={1}>
+            <Typography variant="body2" color="text.secondary">
+              Current Safe
+            </Typography>
+            <EthHashInfo
+              address={pendingTx.safeAddress}
+              name={addressBook[pendingTx.safeAddress]}
+              isAddressBookName={Boolean(addressBook[pendingTx.safeAddress])}
```

Comment:
Nit: you use `addressBook[pendingTx.safeAddress]` a lot. What do you think about extracting it to a variable?